logger_test.go

```diff
@@ -62,43 +61,43 @@ func stubExit() *stubbedExit {
 	return stub
 }
 
-func withJSONLogger(t testing.TB, opts []Option, f func(*jsonLogger, func() []string)) {
+func withJSONLogger(t testing.TB, opts []Option, f func(Logger, *testBuffer)) {
 	sink := newTestBuffer()
 	errSink := newTestBuffer()
 
 	allOpts := make([]Option, 0, 3+len(opts))
 	allOpts = append(allOpts, DebugLevel, Output(sink), ErrorOutput(errSink))
 	allOpts = append(allOpts, opts...)
-	jl := NewJSON(allOpts...)
-	jl.StubTime()
+	logger := newLogger(newJSONEncoder(NoTime()), allOpts...)
 
-	f(jl.(*jsonLogger), func() []string {
-		output := strings.Split(sink.String(), "\n")
-		return output[:len(output)-1]
-	})
-	assert.Empty(t, errSink.String(), "Expected error sink to be empty")
+	f(logger, sink)
+	assert.Empty(t, errSink.Lines(), "Expected error sink to be empty")
 }
 
-func assertMessage(t testing.TB, level, expectedMsg, actualLog string) {
-	expectedLog := fmt.Sprintf(`{"msg":"%s","level":"%s","ts":0,"fields":{}}`, expectedMsg, level)
-	assert.Equal(t, expectedLog, actualLog, "Unexpected log output.")
+func TestJSONLoggerSetLevel(t *testing.T) {
+	withJSONLogger(t, nil, func(logger Logger, _ *testBuffer) {
+		assert.Equal(t, DebugLevel, logger.Level(), "Unexpected initial level.")
+		logger.SetLevel(DebugLevel)
```

Comment:
Not sure how this test slipped in the way it is, but seems like we should be changing the level here instead of using the same value as the initial level?